From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sam Maier <smaier@chromium.org>
Date: Wed, 21 Jun 2023 20:04:37 +0000
Subject: [PATCH] Make compile_resources only use main Android SDK jar

This is blocking the Android U SDK roll as there are robolectric tests,
with their out-of-date robolectric SDK jar, which are failing
compile_resources since we are using attributes which are Android U
exclusive.

Also did some minor cleanup with an unused but similarly named
"android_sdk_dep".

Bug: 1445802
Change-Id: I53ba9721e0330bc6e03b06da197b44fbd1e60427
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4633960
Commit-Queue: Sam Maier <smaier@chromium.org>
Commit-Queue: Mohamed Heikal <mheikal@chromium.org>
Auto-Submit: Sam Maier <smaier@chromium.org>
Reviewed-by: Mohamed Heikal <mheikal@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1160814}
---
 build/config/android/internal_rules.gni |  9 ++++-----
 build/config/android/rules.gni          | 14 --------------
 2 files changed, 4 insertions(+), 19 deletions(-)

diff --git a/build/config/android/internal_rules.gni b/build/config/android/internal_rules.gni
index d3427ebd6c2a6..0b8cf385fdab5 100644
--- a/build/config/android/internal_rules.gni
+++ b/build/config/android/internal_rules.gni
@@ -2149,10 +2149,7 @@ if (enable_java_templates) {
   template("compile_resources") {
     forward_variables_from(invoker, TESTONLY_AND_VISIBILITY)
 
-    _deps = invoker.deps + [
-              invoker.android_sdk_dep,
-              invoker.build_config_dep,
-            ]
+    _deps = invoker.deps + [ invoker.build_config_dep ]
     if (defined(invoker.android_manifest_dep)) {
       _deps += [ invoker.android_manifest_dep ]
     }
@@ -2167,12 +2164,14 @@ if (enable_java_templates) {
     _inputs = [
       invoker.build_config,
       android_sdk_tools_bundle_aapt2,
+      android_sdk_jar,
     ]
 
     _rebased_build_config = rebase_path(invoker.build_config, root_build_dir)
 
+    _rebased_android_sdk_jar = rebase_path(android_sdk_jar, root_build_dir)
     _args = [
-      "--include-resources=@FileArg($_rebased_build_config:android:sdk_jars)",
+      "--include-resources=$_rebased_android_sdk_jar",
       "--aapt2-path",
       rebase_path(android_sdk_tools_bundle_aapt2, root_build_dir),
       "--dependencies-res-zips=@FileArg($_rebased_build_config:deps_info:dependency_zips)",
diff --git a/build/config/android/rules.gni b/build/config/android/rules.gni
index f6206edfd7268..70527997d8a66 100644
--- a/build/config/android/rules.gni
+++ b/build/config/android/rules.gni
@@ -1493,12 +1493,6 @@ if (enable_java_templates && is_android) {
     _non_java_deps = filter_exclude(_invoker_deps, java_target_patterns)
     _java_assetres_deps = [ ":${_java_binary_target_name}__assetres" ]
 
-    if (defined(invoker.alternative_android_sdk_dep)) {
-      _android_sdk_dep = invoker.alternative_android_sdk_dep
-    } else {
-      _android_sdk_dep = default_android_sdk_dep
-    }
-
     # A package name or a manifest is required to have resources. This is
     # added so that junit tests that do not care about the package name can
     # still use resources without having to explicitly set one.
@@ -1538,7 +1532,6 @@ if (enable_java_templates && is_android) {
     compile_resources(_compile_resources_target_name) {
       deps = _non_java_deps + _java_assetres_deps +
              [ ":$_merge_manifest_target_name" ]
-      android_sdk_dep = _android_sdk_dep
       build_config_dep = ":$_build_config_target_name"
       build_config = _build_config
       if (defined(_package_name)) {
@@ -2692,12 +2685,6 @@ if (enable_java_templates && is_android) {
     }
     _generated_proguard_config = "$_base_path.resources.proguard.txt"
 
-    if (defined(invoker.alternative_android_sdk_dep)) {
-      _android_sdk_dep = invoker.alternative_android_sdk_dep
-    } else {
-      _android_sdk_dep = default_android_sdk_dep
-    }
-
     if (defined(_shared_resources_allowlist_target)) {
       _allowlist_gen_dir =
           get_label_info(_shared_resources_allowlist_target, "target_gen_dir")
@@ -2791,7 +2778,6 @@ if (enable_java_templates && is_android) {
 
       build_config = _build_config
       build_config_dep = ":$_build_config_target"
-      android_sdk_dep = _android_sdk_dep
       deps = _java_assetres_deps + _non_java_deps
 
       if (_incremental_apk) {
